src/main/java/com/amazon/corretto/hotpatch/patch/impl/set/PatchSetPatcher.java

```diff
@@ -0,0 +1,177 @@
+
+            boolean patchesApplied = false;
+            for (Class<?> c : instrumentation.getAllLoadedClasses()) {
+                String className = c.getName();
+                for (ClassTransformerHotPatch patch : patches) {
+                    if (patch.isTargetClass(className)) {
+                        logger.log("Patching + " + className + " (" + c.getClassLoader() + ") with patch "
+                                + patch.getName());
+                        try {
+                            instrumentation.retransformClasses(c);
+                            patchesApplied = true;
+                        } catch (UnmodifiableClassException uce) {
+                            logger.log(String.valueOf(uce));
+                        }
+                    }
+                }
+            }
+
+            if (patchesApplied) {
+                logger.log("Vulnerable classes were not found. This agent will continue to run " +
+                        "and transform the target classes if they are loaded. Note that if you have shaded " +
+                        "or otherwise changed the package name for target classes, then this tool may not " +
+                        "find them.");
+            }
+        }
+
+        return Patcher.SUCCESS;
+    }
+
+    @Override
+    public int uninstall() {
+        instrumentation.removeTransformer(transformer);
+        // Retransform after we've removed the transformer to restore the initial class versions.
+        for (Class<?> c : instrumentation.getAllLoadedClasses()) {
+            String className = c.getName();
+            for (ClassTransformerHotPatch patch : getPatches()) {
+                if (patch.isTargetClass(className)) {
+                    logger.log("Un-Patching " + c + " (" + c.getClassLoader() + ") of " + patch.getName());
+                    try {
+                        instrumentation.retransformClasses(c);
+                    } catch (UnmodifiableClassException uce) {
+                        logger.log(String.valueOf(uce));
+                    }
+                }
+            }
+        }
+        logger.log("Uninstalled Patcher: " + getName() + ". Version: " + getVersion());
+        return SUCCESS;
+    }
+
+    /**
+     * A detailed description for this PatchSet that includes the description of the individual patches that will be
+     * applied.
+     * @return A detailed description for this PatchSet that includes the description of the individual patches that
+     *         will be applied.
+     */
+    @Override
+    public String getFullDescription() {
+        StringBuilder sb = new StringBuilder();
+        sb.append(getName()).append("#").append(getVersion()).append(":").append(System.lineSeparator());
+        sb.append("  ").append(getShortDescription()).append(System.lineSeparator());
+        sb.append("  Applies the following patches:").append(System.lineSeparator());
+        for (ClassTransformerHotPatch patch : getPatches()) {
```

Comment:
Should call getEnabledPatches and also list the disabled patches separately.

Reply:
Correct